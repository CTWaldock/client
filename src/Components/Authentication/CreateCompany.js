import React, { Fragment } from 'react'
import Form from '../../UI/Form'
import FormField from '../../UI/FormField'
import Button from '../../UI/Button'
import { StyledLink, ErrorMessage, Center } from './Auth.styles.js'



const CreateCompany = (props) => {
    // console.log(props)
    


    const title = "Lets setup your business huh?"     

    return (
        <Fragment>
            <Center>
                <Form onSubmit={(e) => props.createCompany(e)} title={title}>
                    <FormField label="Company Name" name="name" type="text"  />
                    <FormField label="ABN" name="abn" type="text"  />
                    <FormField label="What type of company do you run?" name="businessType" type="text"  />
                    <FormField label="Company Address" name="address" type="text"  />
                    <FormField label="Company Phone Number" name="phoneNumber" type="text"  />
                    
                    
                    <Button className="formButton" text="Create your company" />
                    {/* <StyledLink to="/register">Need to register a new account?</StyledLink> */}
                </Form>
            </Center>
        </Fragment>
    )
}

export default CreateCompany