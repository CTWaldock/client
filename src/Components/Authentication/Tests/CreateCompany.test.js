import React from 'react'
import CreateCompany from '../CreateCompany'
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom'
// import localStorage from '../../localStorage'

// window.localStorage = localStorage

describe('CreateCompany', () => {
    it('Should render the way we expect', () => {
        
        
        const tree = renderer.create(
            <Router>
                <CreateCompany registerError='something' register='something' url='something' />
            </Router>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
