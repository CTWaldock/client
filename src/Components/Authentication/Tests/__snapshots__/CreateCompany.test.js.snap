// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCompany Should render the way we expect 1`] = `
<div
  className="sc-VigVT gUteem"
>
  <form
    className="sc-gzVnrw eqCBFZ"
    id={undefined}
    onSubmit={[Function]}
  >
    <h3
      className="sc-htoDjs fbZTTv"
    >
      Lets setup your business huh?
    </h3>
    <label
      className="sc-dnqmqq gxITcO"
    >
      Company Name
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="name"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      ABN
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="abn"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      What type of company do you run?
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="businessType"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      Company Address
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="address"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <label
      className="sc-dnqmqq gxITcO"
    >
      Company Phone Number
      <span />
    </label>
    <input
      autoComplete="nope"
      autoFocus={null}
      className="sc-iwsKbI kxDmNQ"
      name="phoneNumber"
      onChange={null}
      placeholder={undefined}
      step={null}
      type="text"
    />
    <button
      className="sc-jTzLTM ffePeW"
    >
      Create your company
    </button>
    <div />
  </form>
</div>
`;
